client/components/Scenario/FinishSlide.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Confirm, Card } from 'semantic-ui-react';
+import './Scenario.css';
+
+class FinishSlide extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isConfirmBoxOpen: this.isScenarioRun ? true : false
+        };
+        this.onCancel = this.onCancel.bind(this);
+        this.onConfirm = this.onConfirm.bind(this);
+    }
+
+    get isScenarioRun() {
+        return location.pathname.includes('/run/');
+    }
+
+    onCancel() {
+        if (this.isScenarioRun) {
+            return;
+        }
+        location.href = this.props.back;
+    }
+
+    onConfirm(event) {
+        if (this.isScenarioRun) {
+            return;
+        }
+        this.setState({ isConfirmBoxOpen: false });
+        this.props.onChange(event, {
+            ended_at: new Date().toISOString()
+        });
+    }
+
+    render() {
+        const { onCancel, onConfirm } = this;
+        const { scenario } = this.props;
+        const { isConfirmBoxOpen } = this.state;
+        const isCardHidden = !isConfirmBoxOpen;
+        const components = scenario.finish && scenario.finish.components;
+        const className = `scenario__card--run${isCardHidden ? '-hidden' : ''}`;
+
+        return (
+            <React.Fragment>
+                <Confirm
+                    content={`If you're ready to finish, click 'I'm done!'`}
+                    header="Ready to finish this scenario?"
+                    cancelButton="Go back"
+                    confirmButton="I'm done!"
+                    onCancel={onCancel}
+                    onConfirm={onConfirm}
+                    open={isConfirmBoxOpen}
+                    size="tiny"
+                />
+                <Card centered className={className}>
+                    <Card.Content style={{ flexGrow: '0' }}>
+                        <Card.Header>Great work!</Card.Header>
```

Comment:
I have a feeling that they'll want to be able to edit this header, but not something we have to address now. 🔮

Reply:
I meant to remove this when `finish` was ...finished 🤣

Reply:
Removed.